Weather App, by Maria Bergman and Vladimir Bykov

vladjnbykov opened the pull request:
Link to our deployed project: https://elegant-davinci-7eedac.netlify.app/

My reflections on how this week's project turned out: Great

Things I'd like to have clarified or explained in more detail: conditional display of a background image (in the body) depending on media queries.

Ulve17 reviewed (approved):
Hi Guys!! Great job with the project this week. 🌈 
Clean and nice code! 😄 
Hopefully we dindn't miss anything, significant.  😨 
Ati, Nazanin and Ulysses 😸

vladjnbykov commented:
Hi guys,
Thank you very much for your comments.
We have now fixed everything and set IDs as camelCases. App works and we 
have now pushed it back to GitHub.

Have a nice day!
Maria and Vlad


------ Originalmeddelande ------
Från: "Ulve17" <notifications@github.com>
Till: "Technigo/project-weather-app" 
<project-weather-app@noreply.github.com>
Kopia: "Vladimir JN Bykov" <vlad.jnbykov@gmail.com>; "Author" 
<author@noreply.github.com>
Skickat: 2021-02-24 20:52:52
Ämne: Re: [Technigo/project-weather-app] Weather App, by Maria Bergman 
and Vladimir Bykov (#163)

>
>@Ulve17 approved this pull request.
>
>Hi Guys!! Great job with the project this week. 🌈
>Clean and nice code! 😄
>Hopefully we dindn't miss anything, significant. 😨
>Ati, Nazanin and Ulysses 😸
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582107461>:
>
> > +const weatherData = () => {
>
>+  fetch(NEW_1URL)
>
>+  .then((response) => {
>
>+    return response.json().then((json) => {
>
>+      weather = json
>
>+      city.innerHTML += weather.name
>
>
>Nice fetch!!!
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582108260>:
>
> > +const dayMonthConverter = (param) => {
>
>+    const date = new Date(param * 1000-3600000)
>
>+    const day = date.getDate()
>
>+    const month = date.getMonth()
>
>+    dayMonth = `${day}/${month+1}`
>
>+    return dayMonth
>
>+}
>
>
>Good code!
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582108821>:
>
> > +      let cityName = weather.name
>
>+      if (cityName === "Stockholm") {
>
>+       coverImage.style.backgroundImage = "url('./images/stockholm.jpg')";
>
>+      } else if (cityName === "Bangkok") {
>
>+        coverImage.style.backgroundImage = "url('./images/bangkok1.jpg')";
>
>+      } else if (cityName === "Hanoi") {
>
>+        coverImage.style.backgroundImage = "url('./images/hanoi.jpg')";
>
>+      } else if (cityName === "Moscow") {
>
>+        coverImage.style.backgroundImage = "url('./images/moscow.jpg')";
>
>+      } else if (cityName === "Malé") {
>
>+        coverImage.style.backgroundImage = "url('./images/male.jpg')";
>
>+      } else if (cityName === "Oslo") {
>
>+        coverImage.style.backgroundImage = "url('./images/oslo.jpg')";
>
>+      } else if (cityName === "Paris") {
>
>+        coverImage.style.backgroundImage = "url('./images/paris.jpg')";
>
>+      } else if (cityName === "Tokyo") {
>
>+        coverImage.style.backgroundImage = "url('./images/tokyo.jpg')";
>
>+      }
>
>
>👍👍👍
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582150064>:
>
> > +const dtable = document.getElementById('dynamic');
>
>+const selectCity = document.getElementById('selectCity');
>
>+let colorWidget = document.getElementById("weather-widget");
>
>+const coverImage = document.getElementById("page-body");
>
>
>One of your id:s is camelCase and some are kebab-case, mabye it is 
>better to have only one of them. 😄
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582151565>:
>
> > +      if (tm <= -10) {
>
>+        colorWidget.style.background = "rgb(209,204,249)";
>
>+      } else if (tm > -10 && tm <= -5) {
>
>+        colorWidget.style.background = "rgb(204,228,249)";
>
>+
>
>+      } else if (tm > -5 && tm <= 0) {
>
>+        colorWidget.style.background = "rgb(204,241,249)";
>
>+
>
>+      } else if (tm > 0 && tm <= 5) {
>
>+        colorWidget.style.background = "rgb(204,249,240)";
>
>+
>
>+      }else if (tm > 5 && tm <= 10) {
>
>+        colorWidget.style.background = "rgb(204,249,218)";
>
>+
>
>+      } else if (tm > 10 && tm <= 15) {
>
>+        colorWidget.style.background = "rgb(216,249,204)";
>
>+
>
>+      } else if (tm > 15 && tm <= 20) {
>
>+        colorWidget.style.background = "rgb(215,237,182)";
>
>+
>
>+      } else if (tm > 20 && tm <= 25) {
>
>+        colorWidget.style.background = "rgb(228,234,161)";
>
>+
>
>+      } else if (tm > 25 && tm <= 30) {
>
>+        colorWidget.style.background = "rgb(249,236,204)";
>
>+
>
>+      } else if (tm > 30) {
>
>+        colorWidget.style.background = "rgb(249,207,204)";
>
>+      }
>
>+    });
>
>+  });
>
>+};
>
>+weatherData();
>
>
>👍
>
>--------------------------------------------------------------------------------
>In code/index.html 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582155172>:
>
> > +<body id="page-body" class="page">
>
>+  <section id="weather-widget" class="weather-widget">
>
>+    <!--Today weather-->
>
>+    <section class="today" id="today">
>
>+      <div id="city" class="city-title"></div><br>
>
>+
>
>+      <div class="now">
>
>+        <div id="dayCurrent"></div>
>
>+        <div id="timeCurrent" class="time-current"></div>
>
>+        <div id="sky"></div>
>
>+        <div id="temperature"></div><br>
>
>+      </div>
>
>+
>
>+      <div id="feelsTemp"></div><br>
>
>+      <div class="sun">
>
>+        <div id="sunrise"></div>
>
>+        <div id="sunset"></div>
>
>+      </div>
>
>+      <div id="sundata" class="sundata"></div>
>
>+    </section>
>
>+
>
>+    <h4 class="f-feels">at noon feels</h4>
>
>+
>
>+    <section class="forecast">
>
>+      <!-- Table for 5 days forecast-->
>
>+      <table id="dynamic" class="forecast-table">
>
>+      </table>
>
>+    </section>
>
>+
>
>+    <h4 class="f-feels">actual temperature at noon</h4>
>
>+
>
>+    <!--Line graph-->
>
>+    <section class="chart">
>
>+      <canvas id="line-chart" width="300" height="200"></canvas>
>
>+    </section>
>
>+
>
>+    <!--City selector: dropdown menu-->
>
>+    <h5 class="select-box">select city</h5>
>
>+
>
>+    <select id="selectCity" class="select-city">
>
>+      <option value="Stockholm,Sweden">Stockholm</option>
>
>+      <option value="Bangkok,Thailand">Bangkok</option>
>
>+      <option value="Hanoi,Vietnam">Hanoi</option>
>
>+      <option value="Moscow,Russia">Moscow</option>
>
>+      <option value="Male,Maldives">Male</option>
>
>+      <option value="Oslo,Norway">Oslo</option>
>
>+      <option value="Paris,France">Paris</option>
>
>+      <option value="Tokyo,Japan">Tokyo</option>
>
>+    </select>
>
>+  </section>
>
>+
>
>+  <script src="./app.js"></script>
>
>
>Here too some id:s are camelCase and other are kebab-case 😸
>
>—
>You are receiving this because you authored the thread.
>Reply to this email directly, view it on GitHub 
><https://github.com/Technigo/project-weather-app/pull/163#pullrequestreview-597675518>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/AQFHVNR7ZEHOSKJKY3DAOXDTAVKJJANCNFSM4X5YH46A>.
>

KarolCislowski commented on code/app.js:
```diff
@@ -0,0 +1,259 @@
+}
+weatherForecast()
+
+/*************************WEATHER GRAPH *******************/
```
Greate that you used a very nice library to create charts 🤓

KarolCislowski reviewed (changes requested):
You created a very nice working app but you need to work some more on styling. Your app does not scale well beyond mobile screen sizes. 🎨

vladjnbykov commented:
Hi Karol,

We have now done some changes to the app which we hope are adressing the 
responsivity issues:

1. Background image was removed from mobile phone view (before it was 
hiding behind the widget)
2. We changed position of the widget and placed it in the center of the 
larger screen
3. We have added a footer for the desktop screen.

Thank you very much,
Best regards,

Maria and Vlad


------ Originalmeddelande ------
Från: "Karol Cislowski" <notifications@github.com>
Till: "Technigo/project-weather-app" 
<project-weather-app@noreply.github.com>
Kopia: "Vladimir JN Bykov" <vlad.jnbykov@gmail.com>; "Author" 
<author@noreply.github.com>
Skickat: 2021-02-25 17:22:29
Ämne: Re: [Technigo/project-weather-app] Weather App, by Maria Bergman 
and Vladimir Bykov (#163)

>
>@KarolCislowski requested changes on this pull request.
>
>You created a very nice working app but you need to work some more on 
>styling. Your app does not scale well beyond mobile screen sizes. 🎨
>
>--------------------------------------------------------------------------------
>In code/app.js 
><https://github.com/Technigo/project-weather-app/pull/163#discussion_r582968674>:
>
> > +        } else if (dw === 6) {
>
>+          dtable.innerHTML += `
>
>+          <tr>
>
>+            <td>Sat ${Math.round(elem.main.feels_like)}°C</td>
>
>+            <td>${elem.weather[0].description}</td>
>
>+            <td><img class="forecast-icon" src="https://openweathermap.org/img/wn/${elem.weather[0].icon}@2x.png"></td>
>
>+          </tr>
>
>+          `
>
>+        }
>
>+      })
>
>+    })
>
>+  })
>
>+}
>
>+weatherForecast()
>
>+
>
>+/*************************WEATHER GRAPH *******************/
>
>
>Greate that you used a very nice library to create charts 🤓
>
>—
>You are receiving this because you authored the thread.
>Reply to this email directly, view it on GitHub 
><https://github.com/Technigo/project-weather-app/pull/163#pullrequestreview-598745823>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/AQFHVNUGD73NNG37DQKFDNDTAZ2MLANCNFSM4X5YH46A>.
>

KarolCislowski reviewed (approved):
Thank you for a fast response 👍🏻 Now it is much better 💯
